### PR TITLE
RMMIS-6514-fix-date-component

### PR DESCRIPTION
**Pull request by kjschulz**
RMMIS-6514: Fixed Date component, allow more than the standard input props for Date, updated tests to support latest changes

### CODE REVIEW

**Comment by kjschulz** [comment]
Weird, fails here but not for me locally (sadpanda)

File: src/Date.jsx [hunk 1]
```diff
@@ -25,22 +25,43 @@ class DateField extends React.Component {
...
+  handleDateChange(date, dateString) {
+    let _div = React.findDOMNode(this);
+    let event = new Event('change', {bubbles: true});
+    event.date = date;
+    event.dateString = dateString;
+    _div.dispatchEvent(event);
+  }
+
+  handleManualChange(e) {
+    let value = {date: _.clone(e.date), dateString: _.clone(e.dateString)};
```
Comment (mallenscdhhs) [inline src/Date.jsx]: why is `value` an object here? shouldn't it be a string?

File: src/Date.jsx [hunk 2]
```diff
@@ -25,22 +25,43 @@ class DateField extends React.Component {
...
     };
   }
 
   render() {
-    let props = _.pick(this.props, this.props.inputProps);
     return (
-      <div onChange={this.handleDateChange}>
-        <DateTimePicker {...props}
-          value={DateField.getDateValue(this.props.value)} />
+      <div>
```
Comment (mallenscdhhs) [inline src/Date.jsx]: do we need this extra `<div>` here?

Comment (kjschulz) [inline src/Date.jsx]: Nope, will update.

**Pull request merged**